draft-ietf-quic-multipath.md

```diff
@@ -455,23 +521,65 @@ the server MAY wait for a short, limited time such as one PTO if a path
 probing packet is received on a new path before sending the
 CONNECTION_CLOSE frame.
 
+Note that PATH_ABANDON frame is also used as a signal for the retirement 
+of the associated Path Identifier. When endpoint received PATH_ABANDON frame,
+it SHOULD not use the associated Path Identifier in future packets, it can 
+only use the Path ID in ACK_MP frames for inflight packets or 
+in MP_RETIRE_CONNECTION_ID frames for CID retirement.
+
 ### Refusing a New Path
 
 An endpoint may deny the establishment of a new path initiated by its
 peer during the address validation procedure. According to
 {{QUIC-TRANSPORT}}, the standard way to deny the establishment of a path
 is to not send a PATH_RESPONSE in response to the peer's PATH_CHALLENGE.
 
-### Effect of RETIRE_CONNECTION_ID Frame {#retire-cid-close}
+If endpoint fails to validate a path and consume an available Path Identifier,
+it SHOULD send a PATH_ABANDON frame to retire the associated Path Identifier. 
```

Comment:
```suggestion
A failed path validation consumes the Path ID used for probing of this path.
An endpoint MUST not use the same Path ID to probe a different path. Instead, it
MUST send a PATH_ABANDON frame to retire the Path ID. 
```

Reply:
Wouldn't it be enough to retire the connection ID? Abandoning the whole (potential) path seems like overkill.

Reply:
I think that complicates things. Path validation can fail for a number of reasons. In particular, either the forward path (client -> server) or the return path (server -> client) could be broken separately.

Therefore, from the fact that path validation failed, you cannot conclude that the server didn't see that particular path ID. Nuking the entire path ID seems like the easiest way to avoid confusion between client and server.